server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/PrimaryAllocationIT.java

```diff
@@ -359,22 +367,26 @@ public void testForceStaleReplicaToBePromotedForGreenIndex() {
         ensureGreen();
         final String nodeWithoutData = randomFrom(dataNodes);
         final int shardId = 0;
-        IllegalArgumentException iae = expectThrows(
-            IllegalArgumentException.class,
-            clusterAdmin().prepareReroute().add(new AllocateStalePrimaryAllocationCommand(idxName, shardId, nodeWithoutData, true))
+        assertEquals(
+            "[allocate_stale_primary] primary [" + idxName + "][" + shardId + "] is already assigned",
+            ClusterRerouteUtils.expectRerouteFailure(
```

Comment:
Another exception type check that wasn't carried over from the old code.

Reply:
Thanks, see 68ebfe8.